package/MDAnalysis/bundle.py

```diff
@@ -0,0 +1,267 @@
+
+    When providing filenames, if additional keyword arguments are required to
+    load the corresponding Universe, these can be provided as *univ_kwargs*. 
+    Only one set of *univ_kwargs* may be provided, assumed to be common for all;
+    otherwise, simulations should be loaded separately before bundling or 
+    added separately using :func:`add_to_bundle`.
+
+    If not passing in Sims, these are created, using names passed in with the
+    *names* kwarg (as a list). If names are not provided, the index is 
+    used instead. 
+
+    Metadata and auxiliary data may be added on initialisation using the *data*  
+    and *auxs* kwargs, respectively, passing in the (ordered) list of 
+    values/auxiliary filenames etc for each simulation keyworded by metadata/auxiliary
+    name; for metadata where all simulations have the same value, a single value
+    may be gien in place of a list. e.g.::
+
+        make_bundle([Sim1, Sim2], auxs={'pull_f': ['sim1.xvg', 'sim2.xvg']},
+                    data = {'dist':[1,2], 'temp':300})
+
+    If additional kwargs are requried for adding auxiliaries, these may 
+    be supplied with *aux_kwargs*::
+
+        make_bundle([Sim1, Sim2], auxs={'pull_f': ['sim1.xvg', 'sim2.xvg']},
+                     aux_kwargs={'pull_f': {'dt': 400}})
+
+    Parameters
+    ----------
+    simulation : lst
+        list of simulation to add (ordered if required) [[ list possible types? ]]
+    topology : (str, lst), optional
+        filename for single topology corresponding to all simulations for which
+        only a trajectory filename is provided, or ordered list of topology
+        filenames corresponding to each simulation
+    univ_kwargs : dict, optional
+        dictionary of additional arguments to pass to Universe for each 
+        simulation passed as filenames
+    names : lst of str
+        ordered list of names for the simulation
+    data : dict
+        dictionary of metadata names and corresponding values to add
+    auxs : dixt
+        dictionary of auxiliary names and corresponding data to add
+    aux_kwargs : dict of dict
+        dictionary of additional arguments to use when loading each set of
+        auxiliary data
+    """
+    if names is None:
+        # if names not provided, use the index for name
+        names = map(str, range(len(simulations)))
+    elif len(names) != len(simulations):
+        raise ValueError("Number of simulations ({}) and number of names ({}) do "
+                         "not match".format(len(simulations), len(names)))
+
+    if not isinstance(topology, (list, np.ndarray)):
+        # assume all use the same topology
+        tops = [topology]*len(simulations)
+    elif len(topology) != len(simulations):
+        raise ValueError("Number of simulations ({}) and number of topologies ({}) "
+                         "do not match".format(len(simulations), len(topology)))
+
+    bundle = dtr.Bundle()
+    for i, sim in enumerate(simulations):
```

Comment:
You could use `for sim, top, name in zip(simulations, tops, names)` rather than an index. `zip`, like `range` and `map`, is a generator on python 3. To avoid surprises you should import `from six.moves import zip`.